docker/Dockerfile

```diff
@@ -0,0 +1,24 @@
+FROM ruby:2.7.1
+
+ARG RAILS_ENV
+ENV RAILS_ENV=$RAILS_ENV
+
+RUN apt-get update -qq && \
+  apt-get install -y --no-install-recommends gnupg2 curl && \
+  apt-get clean && \
+  rm -rf /var/lib/apt/lists/*
+
+ENV APP_HOME /app/src
+WORKDIR $APP_HOME
+
+ADD ./ $APP_HOME/
+
+ENV BUNDLE_GEMFILE=$APP_HOME/Gemfile \
+  BUNDLE_JOBS=2 \
+  BUNDLE_PATH=/bundle
+RUN bundle install
```

Comment:
Can you describe the image build penalty involved in having `bundle install` run after all of the source code is copied in?

Reply:
It takes longer to build the image, however, if your dependencies don't change very often, then you only have to pay that penalty when you're building it the first time. The compose file mounts a shared docker volume that is mounted up. You could bundle all your gems and commit them to the repo, but then the repository size grows dramatically.